### PR TITLE
Newsletters feature for Ghost 5.0

### PR DESCRIPTION
**Pull request by sam-lord**
Added the newsletter feature for Ghost 5.0.

In the form, a new dropdown allows the user to select either single or multiple newsletter mode. Single newsletter mode uses the original "subscribed" boolean, and multiple newsletter mode includes a dropdown to select the newsletters.

### CODE REVIEW

**Review by naz — CHANGES_REQUESTED** [review]
@sam-lord we have kept an ok-ish condition of our test suites for this integration. I think it would be way easier for me to review the changes (especially all the pesky ifs) if we added coverage :ok_hand:

**Review by naz — APPROVED** [review]
I think this should be good to go! I've also had a look if instead of having a hidden trigger for `newsletter_created` we could have utilized a "resource" (just because we've never used it before). That seemed like a dead end as there doesn't seem to be a way to hide the definition like with triggers (or it's not well documented :smile:). Have you considered this approach maybe?

Also, another slightly confusing thing I've noticed was this extra fields warning, after playing with the newsletter dropdown combinations too much. Doesn't seem to be any straight forward way to clean up this property though?
Screenshot from 2022-05-18 13-22-29

Tested with 4.6 and 5.0 Ghost instances. Feel free to merge in! :heavy_check_mark:

**Comment by sam-lord** [comment]
Thanks for the review. Before merging I wanted to respond to these :)

The resources type looks really useful, I think if we do a breaking change at some point it would be good to replace those triggers that are only used for dynamic dropdowns. Hadn't considered at the time of writing these since I wanted the changes to be as low-impact and blend in as much as possible.

Extra fields warning is a little annoyance, it appears if you go through the process of making a Zap using the "single newsletter" mode then change to multiple newsletters, or vice versa. Fortunately it doesn't matter whether the user clicks to remove the extra fields, the Zap will still work. I wonder if there's a hack we can do to make Zapier think we're using every field -- looping over them to make a hash or something 😂

**Pull request merged**
